features/step_definitions/logging.rb

```diff
@@ -705,3 +705,135 @@
+    | f | #{BushSlicer::HOME}/testdata/logging/clusterlogforwarder/kafka/amq/03_sub_amqstreams_template.yaml |
+    | p | AMQ_NAMESPACE=#{project_name} |
+  })
+  raise "Error subscript amqstreams" unless @result[:success]
+
+  step %Q/a pod becomes ready with labels:/, table(%{
+    | name=amq-streams-cluster-operator |
+  })
+
+  step %Q/I process and create:/, table(%{
+    | f | #{BushSlicer::HOME}/testdata/logging/clusterlogforwarder/kafka/amq/04_kafka_my-cluster_amqstreams_template.yaml |
+    | p | AMQ_NAMESPACE=#{project_name} |
+  })
+  raise "Error create kafka" unless @result[:success]
+
+  @result = admin.cli_exec(:create, f: "#{BushSlicer::HOME}/testdata/logging/clusterlogforwarder/kafka/amq/05_kafkatopics_amqstreams.yaml")
+  raise "Error create kafka topics" unless @result[:success]
+
+  step %Q/3 pods becomes ready with labels:/, table(%{
+    | strimzi.io/name=my-cluster-kafka |
+  })
+  raise "Error kafka cluster not ready" unless @result[:success]
+end
+
+# Get some records from a kafka topic 
+# https://datacadamia.com/dit/kafka/kafka-console-consumer
+# it seems some options are not supported to 'oc run' command. Instead of that, I use a job to view logs, and delete it later. Leave the 'oc run' steps here for later
+#oc run kafka-consumer -ti --image=registry.redhat.io/amq7/amq-streams-kafka-25-rhel7:1.5.0 --rm=true --restart=Never -- bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic topic-logging-app --max-messages 10 --from-beginning
+#step %Q/I run the :run client command with:/,table(%{
+#    | name    | kafka-consumer |
+#    | image   | #{kafka_image} |
+#    | restart | Never |
+#    | command | true  |
+#    | cmd     | bin/kafka-console-consumer.sh  |
+#    | cmd     | --bootstrap-server my-cluster-kafka-bootstrap:9092  |
+#    | cmd     | --topic #{topic_name}   |
+#    | cmd     | --from-beginning  |
+#    | cmd     | --max-messages #{record_num} |
+#    | cmd     | --timeout-ms=300000  |
+#})
+Given /^I get(?: (\d+))? records from the #{QUOTED} kafka topic$/ do | record_num, topic_name|
+  record_num = record_num ? record_num.to_str : "10"
+  job_name=topic_name
+  kafka_image=stateful_set('my-cluster-kafka').containers_spec(user: user)[0].image
+  retries = 3
+  while true
+    step %Q/I process and create:/,table(%{
+      | f | #{BushSlicer::HOME}/testdata/logging/clusterlogforwarder/kafka/amq/21_job_topic_consumer_from_beginning_template.yaml |
+      | p | KAFKA_IMAGE=#{kafka_image} |
+      | p | KAFKA_TOPIC=#{topic_name}  |
+      | p | MAX_MESSAGES=#{record_num} |
+      | p | JOB_NAME=#{job_name} |
+    })
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | job-name=#{job_name}|
+    })
+    sleep 30
+    step %Q/I run the :logs client command with:/, table(%{
+      | resource_name | #{ pod.name} |
+    })
+
+    admin.cli_exec(:delete, object_type: 'job', object_name_or_id: job_name, n: cb.kafka_project.name)
```

Comment:
If any one of the above steps failed, this step couldn't be executed, then the whole step will fail.